### PR TITLE
Add cmocka 1.1.5

### PR DESCRIPTION
**Pull request by shuryanc**
This is to add comcka 1.1.5, which provides a testing framework for C.

### CODE REVIEW

File: components/library/comcka/Makefile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+COMPONENT_ARCHIVE=      $(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_HASH=	sha256:51eba78277d51f0299617bedffc388b2b4ea478f5cc9876cc2544dae79638cb0
+COMPONENT_ARCHIVE_URL=	https://gitlab.com/cmocka/cmocka/-/archive/$(COMPONENT_NAME)-$(COMPONENT_VERSION)/$(COMPONENT_ARCHIVE)
+COMPONENT_LICENSE=      APACHE20
+COMPONENT_LICENSE_FILE= COPYING     
+
+include $(WS_MAKE_RULES)/prep.mk
+include $(WS_MAKE_RULES)/cmake.mk
+include $(WS_MAKE_RULES)/ips.mk
+
```
Comment (AndWac) [inline components/library/comcka/Makefile]: Please read doc/current-tasks.md and change your Makefile to the new style, eg. use BUILD_BITS=64, BUILD_STYLE=cmake and replace the includes accordingly.

File: components/library/comcka/Makefile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#
+# Copyright 2022, Daniel Chan
+#
+
+include ../../../make-rules/shared-macros.mk
+
+BUILD_BITS=		64
+BUILD_STYLE=cmake
+COMPONENT_NAME=         cmocka
```
Comment (AndWac) [inline components/library/comcka/Makefile]: This should be above the first include.

File: components/library/comcka/Makefile [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+COMPONENT_PROJECT_URL=  https://cmocka.org/
+COMPONENT_FMRI=         library/cmocka
+COMPONENT_CLASSIFICATION=System/Libraries
+COMPONENT_SUMMARY=      Unit testing framework for C with support for mock objects
+COMPONENT_SRC=          $(COMPONENT_NAME)-$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=      $(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_HASH=	sha256:51eba78277d51f0299617bedffc388b2b4ea478f5cc9876cc2544dae79638cb0
+COMPONENT_ARCHIVE_URL=	https://gitlab.com/cmocka/cmocka/-/archive/$(COMPONENT_NAME)-$(COMPONENT_VERSION)/$(COMPONENT_ARCHIVE)
+COMPONENT_LICENSE=      APACHE20
+COMPONENT_LICENSE_FILE= COPYING
```
Comment (AndWac) [inline components/library/comcka/Makefile]: Elsewhere we have used Apache 2.0

**Comment by mtelka** [comment]
Is this comcka or cmocka?

**Comment by AndWac** [comment]
Please fix the folder name and also add the sample-manifest.p5m.

**Comment by AndWac** [comment]
We already have developer/ceck as a C test framework. You should also move this into the developer folder where it is better suited. It should also have COMPONENT_CLASSIFICATION=Development/C
like the check framework has.
Finally, when everything is fine, please squash the commits and only leave an appropriate commit message.

**Comment by AndWac** [comment]
Your PR is empty now.

**Comment by AndWac** [comment]
Please squash the commits (and remove unnecessary commit messages).

**Comment by shuryanc** [comment]
Rebased.

**Pull request merged**